core/src/main/java/org/elasticsearch/common/collect/CopyOnWriteHashMap.java

```diff
@@ -511,12 +516,12 @@ public int size() {
      * Same as {@link #copyAndPut(Object, Object)} but for an arbitrary number of entries.
      */
     public CopyOnWriteHashMap<K, V> copyAndPutAll(Map<? extends K, ? extends V> other) {
-        return copyAndPutAll(other.entrySet());
+        return copyAndPutAll(other.entrySet().stream());
```

Comment:
If you like the change [proposed](https://github.com/elastic/elasticsearch/pull/13533/files#r39337198) on `copyAndPutAll`, the call to `Collection#stream` isn't necessary.